pinot-core/src/main/java/org/apache/pinot/core/data/manager/BaseTableDataManager.java

```diff
@@ -644,12 +659,13 @@ private File downloadAndStreamUntarWithRateLimit(String segmentName, SegmentZKMe
   }
 
   @VisibleForTesting
-  File untarAndMoveSegment(String segmentName, File tarFile, File tempRootDir)
+  protected File untarAndMoveSegment(String segmentName, File tarFile, File tempRootDir)
       throws IOException {
     File untarDir = new File(tempRootDir, segmentName);
     try {
       // If an exception is thrown when untarring, it means the tar file is broken
       // or not found after the retry. Thus, there's no need to retry again.
+      // unless untar retry is configured, which will retry this function call
```

Comment:
The original description says that there is no need to retry, implying that if it fails once, retries will also fail. In which cases a retry will be useful?

Reply:
That is the premise of this issue: https://github.com/apache/pinot/issues/9152 for which this PR is raised.
In case of partial/corrupted tar downloads the untar fails, so the download and untar is retried.
This function is externally retried, so should I remove this comment ?